From ddc33e0728ebc6ea66999edbf12fd5f2e3ddd548 Mon Sep 17 00:00:00 2001
From: Keith Maxwell <keith.maxwell@gmail.com>
Date: Sun, 29 Mar 2015 22:21:29 +0100
Subject: [PATCH] Increase maximum page height & width in canon_dr

The Canon P-215 supports pages up to 356mm long (~14in). Before this change
the option for the height of the scan is limited to 11in (~279mm) the
height of US-letter paper as set up in backend/canon_dr.c:1653:

    -y 0..279.364mm (in steps of 0.0211639) [279.364]
        Height of scan-area.

This patch changes that range to:

    -y 0..359.11mm (in steps of 0.0211639) [279.364]
        Height of scan-area.

Reference specification:
http://www.canon.co.uk/for_work/products/document_imaging_systems/high_speed_document_scanners/p-215/
---
 backend/canon_dr.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/backend/canon_dr.c b/backend/canon_dr.c
index 0b120fa..5cee00f 100644
--- a/backend/canon_dr.c
+++ b/backend/canon_dr.c
@@ -1962,7 +1962,7 @@ sane_get_option_descriptor (SANE_Handle handle, SANE_Int option)
     /* values stored in 1200 dpi units */
     /* must be converted to MM for sane */
     s->tl_x_range.min = SCANNER_UNIT_TO_FIXED_MM(s->min_x);
-    s->tl_x_range.max = SCANNER_UNIT_TO_FIXED_MM(get_page_width(s));
+    s->tl_x_range.max = SCANNER_UNIT_TO_FIXED_MM(s->max_x);
     s->tl_x_range.quant = MM_PER_UNIT_FIX;
   
     opt->name = SANE_NAME_SCAN_TL_X;
@@ -1980,7 +1980,7 @@ sane_get_option_descriptor (SANE_Handle handle, SANE_Int option)
     /* values stored in 1200 dpi units */
     /* must be converted to MM for sane */
     s->tl_y_range.min = SCANNER_UNIT_TO_FIXED_MM(s->min_y);
-    s->tl_y_range.max = SCANNER_UNIT_TO_FIXED_MM(get_page_height(s));
+    s->tl_y_range.max = SCANNER_UNIT_TO_FIXED_MM(s->max_y);
     s->tl_y_range.quant = MM_PER_UNIT_FIX;
   
     opt->name = SANE_NAME_SCAN_TL_Y;
@@ -1998,7 +1998,7 @@ sane_get_option_descriptor (SANE_Handle handle, SANE_Int option)
     /* values stored in 1200 dpi units */
     /* must be converted to MM for sane */
     s->br_x_range.min = SCANNER_UNIT_TO_FIXED_MM(s->min_x);
-    s->br_x_range.max = SCANNER_UNIT_TO_FIXED_MM(get_page_width(s));
+    s->br_x_range.max = SCANNER_UNIT_TO_FIXED_MM(s->max_x);
     s->br_x_range.quant = MM_PER_UNIT_FIX;
   
     opt->name = SANE_NAME_SCAN_BR_X;
@@ -2016,7 +2016,7 @@ sane_get_option_descriptor (SANE_Handle handle, SANE_Int option)
     /* values stored in 1200 dpi units */
     /* must be converted to MM for sane */
     s->br_y_range.min = SCANNER_UNIT_TO_FIXED_MM(s->min_y);
-    s->br_y_range.max = SCANNER_UNIT_TO_FIXED_MM(get_page_height(s));
+    s->br_y_range.max = SCANNER_UNIT_TO_FIXED_MM(s->max_y);
     s->br_y_range.quant = MM_PER_UNIT_FIX;
   
     opt->name = SANE_NAME_SCAN_BR_Y;
-- 
2.3.0

